taichi/ir/frontend_ir.cpp

```diff
@@ -925,18 +925,19 @@ void SNodeOpExpression::flatten(FlattenContext *ctx) {
   } else if (op_type == SNodeOpType::get_addr) {
     ctx->push_back<SNodeOpStmt>(SNodeOpType::get_addr, snode, ptr, nullptr);
   } else if (op_type == SNodeOpType::append) {
-    flatten_rvalue(value, ctx);
-
+    for (auto &value : values) {
+      flatten_rvalue(value, ctx);
+    }
     auto alloca = ctx->push_back<AllocaStmt>(PrimitiveType::i32);
     auto addr =
         ctx->push_back<SNodeOpStmt>(SNodeOpType::allocate, snode, ptr, alloca);
-    auto ch_addr = ctx->push_back<GetChStmt>(addr, snode, 0);
-    ctx->push_back<GlobalStoreStmt>(ch_addr, value->stmt);
+    for (int i = 0; i < values.size(); i++) {
+      auto ch_addr = ctx->push_back<GetChStmt>(addr, snode, i);
+      ctx->push_back<GlobalStoreStmt>(ch_addr, values[i]->stmt);
```

Comment:
Do we need to verify whether `ret_type` of `values[i]` fits with that of the `snode`, in `SNodeOpExpression::type_check`? (with type_promotion considered)

Reply:
This is done inside the type checking of `GlobalStoreStmt` so we don't need to check it inside `SNodeOpExpression::type_check`?